File: check_batches.py

import os
from supabase import create_client, Client
from dotenv import load_dotenv

load_dotenv()

url = os.environ.get("VITE_SUPABASE_URL")
key = os.environ.get("VITE_SUPABASE_ANON_KEY")

if not url or not key:
    print("Error: Missing env vars")
    exit(1)

supabase: Client = create_client(url, key)

print("Fetching honey_batches...")
try:
    response = supabase.table("honey_batches").select("*").execute()
    batches = response.data
    print(f"Found {len(batches)} batches.")
    for b in batches:
        print("---")
        print(f"ID: {b.get('id')}")
        print(f"Farmer: {b.get('farmer_name')}")
        print(f"Beekeeper: {b.get('beekeeper_name')}") # Check if this column exists
        print(f"Region: {b.get('location_region')}") # Check if this column exists
        print(f"Keys: {b.keys()}")
except Exception as e:
    print(f"Error: {e}")

File: check_batches.js
import { createClient } from '@supabase/supabase-js';

const supabaseUrl = 'https://lqdxsgnoeickomhsgeco.supabase.co';
const supabaseKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImxxZHhzZ25vZWlja29taHNnZWNvIiwicm9zZSI6ImFub24iLCJpYXQiOjE3Njc4NTIwMDEsImV4cCI6MjA4MzQyODAwMX0.UtOsTUJS2Z8uqi7pzDqn8eZdv9T5CvZRVGDsOrC_gms';

const supabase = createClient(supabaseUrl, supabaseKey);

async function checkColumns() {
    console.log("Fetching first row from 'batches'...");
    const { data, error } = await supabase.from('batches').select('*').limit(1);
    if (error) {
        console.error(error.message);
    } else {
        console.log("Columns in 'batches':", Object.keys(data[0] || {}));
        console.log("Sample row:", data[0]);
    }
}

checkColumns();
